akvo/rsr/static/scripts-src/my-results/selectors.js

```diff
@@ -133,7 +133,47 @@ export const getIndicatorsAggregateActualValue = createSelector(
             const aggregateValue = childPeriodIds[indicatorId].reduce((acc, periodId) => {
                 return acc + actualValue[periodId];
             }, 0);
-            return {...acc, [indicatorId]: aggregateValue}
+            return {...acc, [indicatorId]: aggregateValue};
+        }, {});
+    }
+);
+
+
+export const getIndicatorsAggregateTargetValue = createSelector(
+    /*
+      Return an object on the form:
+      {indicatorId1: <aggregateTargetValue1>, indicatorId2: <aggregateTargetValue2>,...}
+    */
+    [getIndicatorIds, getIndicatorsChildrenIds, getPeriodObjects],
+    (indicatorIDs, childPeriodIds, periodObjects) => {
+        return indicatorIDs.reduce((acc, indicatorId) => {
+            const aggregateValue = childPeriodIds[indicatorId].reduce((acc, periodId) => {
```

Comment:
This _might perhaps_ blow up if childPeriodIds is ever undefined. The problem is that I'm not sure. If you look further down at **getNeedReportingPeriods** for example I do this
```javacript
    ...
    unlockedPeriods && updateObjects && unlockedPeriods.filter(
    ...
```
to make sure things don't go south if the params are undefined. In the case of **getNeedReportingPeriods** this could happen if you opened the tree before all models were loaded (I think, it might have been some of the other selectors, but you get the drift). How to handle uninitialized objects and arrays is a general headache that I don't have a good solution for. It's probably something we should try to solve though.